internal/kafka/test/integration/kafka_expiration_test.go

```diff
@@ -0,0 +1,513 @@
+
+	kafkaRequestList, resp, err := client.DefaultApi.GetKafkas(orgCtx, &public.GetKafkasOpts{})
+	if resp != nil {
+		_ = resp.Body.Close()
+	}
+	g.Expect(err).ToNot(gomega.HaveOccurred())
+
+	// verify that kafkas with an expired quota and grace period is suspended
+	// other kafkas should not have an expires_at value set
+	for _, k := range kafkaRequestList.Items {
+		if strings.HasPrefix(k.Name, "kafka-to-suspend-") {
+			_, err := common.WaitForKafkaToReachStatus(orgCtx, test.TestServices.DBFactory, client, k.Id, constants.KafkaRequestStatusSuspended)
+			g.Expect(err).NotTo(gomega.HaveOccurred(), "error waiting for kafka %q to be suspended", k.Id)
+			g.Expect(k.ExpiresAt).NotTo(gomega.BeNil())
+			continue
+		}
+		g.Expect(k.ExpiresAt).To(gomega.BeNil())
+	}
+
+	// test cases for verifying expiration logic for users registered as service account
+	// create dummy kafkas for user registered as a service account in the quota management list
+	for i, k := range kafkas {
+		k.ID = fmt.Sprintf("dummy-kafka-for-user-%d", i)
+		k.OrganisationId = "unregistered-org"
+		k.Owner = dummyUser1
+	}
+
+	if err := db.Create(&kafkas).Error; err != nil {
+		g.Expect(err).NotTo(gomega.HaveOccurred())
+		return
+	}
+
+	saAccount := h.NewAccount(dummyUser1, "", "", "unregistered-org")
+	saCtx := h.NewAuthenticatedContext(saAccount, nil)
+
+	kafkaDeletionErr = common.WaitForNumberOfKafkaToBeGivenCount(saCtx, test.TestServices.DBFactory, client, int32(len(kafkasToRemain)), func(builder common.PollerBuilder) common.PollerBuilder {
+		return builder.DumpDB("kafka_requests", "", "id", "name", "status", "actual_kafka_billing_model", "expires_at")
+	})
+	g.Expect(kafkaDeletionErr).NotTo(gomega.HaveOccurred(), "Error waiting for kafka deletion: %v", kafkaDeletionErr)
+
+	kafkaRequestList, resp, err = client.DefaultApi.GetKafkas(saCtx, &public.GetKafkasOpts{})
+	if resp != nil {
+		_ = resp.Body.Close()
+	}
+	g.Expect(err).ToNot(gomega.HaveOccurred())
+
+	// verify that kafkas with an expired quota and grace period is suspended
+	// other kafkas should not have an expires_at value set
+	for _, k := range kafkaRequestList.Items {
+		if strings.HasPrefix(k.Name, "kafka-to-suspend-") {
+			_, err := common.WaitForKafkaToReachStatus(saCtx, test.TestServices.DBFactory, client, k.Id, constants.KafkaRequestStatusSuspended)
+			g.Expect(err).NotTo(gomega.HaveOccurred(), "error waiting for kafka %q to be suspended", k.Id)
+			g.Expect(k.ExpiresAt).NotTo(gomega.BeNil())
+			continue
+		}
+		g.Expect(k.ExpiresAt).To(gomega.BeNil())
+	}
+}
+
+func TestKafka_ExpirationBasedOnLifespanSeconds(t *testing.T) {
+	g := gomega.NewWithT(t)
+
+	ocmServer := mocks.NewMockConfigurableServerBuilder().Build()
+	defer ocmServer.Close()
+
+	longLifespanSeconds := 178000
+	shortLifespanSeconds := 2
+
+	h, client, tearDown := test.NewKafkaHelperWithHooks(t, ocmServer, func(kafkaConfig *config.KafkaConfig, quotaManagementListConfig *quota_management.QuotaManagementListConfig) {
```

Comment:
do we need to set `quotaManagementListConfig.EnableInstanceLimitControl` to `true`?

Reply:
don't really need to, Kafka expiration isn't really based on instance limit with quota management list, rather it's based on if the user has the quota to create that Kafka and if that quota is still active/expired (configured by expiration_date in the kafka supported types config).

Reply:
I pointed this out as there were recently failures in nightly (in the original TestKafka_KafkaExpiration fixed by https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1555) related to expiration since instance limit control is turned off: https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/e63f0cf8637886adf0df7d29dac13c34106f4857/internal/kafka/internal/environments/development.go#L20

Due to that, quota was always considered active https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/e63f0cf8637886adf0df7d29dac13c34106f4857/internal/kafka/internal/services/quota/quota_management_list_service.go#L236-L238 

quota being active or not is used in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/e63f0cf8637886adf0df7d29dac13c34106f4857/internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go#L418 to reset the `expires_at` to nil in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/e63f0cf8637886adf0df7d29dac13c34106f4857/internal/kafka/internal/workers/kafka_mgrs/kafkas_mgr.go#L430-L433

This behavior might have some on impacts on the Kafka that we expect to be deleted since they've expired https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/e63f0cf8637886adf0df7d29dac13c34106f4857/internal/kafka/test/integration/kafka_expiration_test.go#L284-L332

Reply:
TL;DR:

If the test passes when OCM_ENV is set to `development` then we are good. 
If it fails, please let's consider the suggestion I raised above for both of these tests.

Reply:
Thanks @machi1990, good catch on this, I completely missed that

https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/e63f0cf8637886adf0df7d29dac13c34106f4857/internal/kafka/internal/services/quota/quota_management_list_service.go#L236-L238

It makes sense to set that parameter to true in this case. 👍

Reply:
This makes me wonder: Why are we considering the quota being active when instance limit control is disabled?

Reply:
@miguelsorianod that's true. Not sure if it makes sense. no matter if there's a limit control, the quota being active should really depend on whether the quota is given to the org/user and if the expiration date hasn't been reached when it's defined. 

I assume this was added to mimic the behaviour of the ReserveQuota where it skips the quota allowance check (whether org/user is given quota) + instance limit checks (whether org/user has reached their quota limit) when this flag is set to false. 

The name of the flag is misleading if this is the expected behaviour. cc @ziccardi

Reply:
+1

Reply:
I think this needs further discussion to figure out what the expected behaviour is. Let's not include this change with this PR. If we decide that this behaviour/parameter name needs to be changed, I can create a follow on PR with those changes.

Reply:
@miguelsorianod @JameelB the reason is that if the `EnableInstanceLimitControl` is `false`, then the user has an unlimited quota. With the previous version of the quota list, the user could create unlimited instances (disregarding the billing model)  if it was false, so the same is done here.

Reply:
I think we should reevaluate this and the implications.
Conceptually it seems it might be independent. Maybe is the name that is confusing as Jameel mentioned?